### DISCUSSION TITLE
Retry failed steps

### DISCUSSION THREAD

**Discussion by bennypowers** [community/community#12933] [Actions]
I'd like actions to automatically retry steps:

```yml
  - name: Test
    run: make ci
    with:
      retries: 3
```

**Comment by ethomson** [comment]
Interesting! I hadn't considered adding retries at this level (one of my projects _does_ have some integration tests that hit the network and thus does have a retry loop). There are definitely a handful of things to consider here - in this example, since there's no clean-up at the _step_ level and you'd start the retry from the state where the failed job left off - but it's an interesting idea nonetheless. I'll make sure that this is on the backlog.

Thanks much for the feedback @bennypowers, we appreciate it.
